src/OpenLoco/src/Windows/News/News.cpp

```diff
@@ -536,8 +541,8 @@ namespace OpenLoco::Ui::Windows::NewsWindow
                     auto vehicle = EntityManager::get<Vehicles::VehicleHead>(EntityId(itemIndex));
                     if (vehicle == nullptr)
                     {
-                        WindowManager::close(WindowType::news);
-                        break;
+                        // We could close the window now but then we can't view old news about removed vehicles.
+                        return;
```

Comment:
This is the main fix by breaking it would enter code that would tend to cause a crash.